src/test/java/core/basesyntax/service/impl/DataParsedImplTest.java

```diff
@@ -0,0 +1,62 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ParserService;
+import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class DataParsedImplTest {
+    private static List<String> data;
+    private static FruitTransaction first;
+    private static FruitTransaction last;
+    private final ParserService parserService = new DataParsedImpl();
+    private List<FruitTransaction> fruitTransactionList;
+    private List<FruitTransaction> emptyList;
+
+    @BeforeAll
+    static void beforeAll() {
+        data = List.of("    type,fruit,quantity",
+                "    b,banana,20",
+                "    b,apple,100",
+                "    s,banana,100");
+        first = new FruitTransaction(FruitTransaction.Operation.BALANCE, "banana", 20);
+        last = new FruitTransaction(FruitTransaction.Operation.SUPPLY, "banana", 100);
+    }
+
+    @BeforeEach
+    void setUp() {
+        emptyList = parserService.parseData(Collections.emptyList());
+        fruitTransactionList = parserService.parseData(data);
+    }
```

Comment:
It's unconventional to run the method under test in a setup method (like @BeforeEach) and then use its results in multiple test methods. Typically, each test should call the method under test independently to ensure isolation between tests. This approach allows each test to be more focused, understandable, and maintainable